docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -118,6 +121,117 @@ def my_job():
     op_2(op_1())
 ```
 
+### Applying IO managers to assets
+
+By default, materializing an asset will pickle it to a local file named "my_asset", in a temporary directory. You can specify this directory by providing a value for the `local_artifact_storage` property in your dagster.yaml file.
+
+[IO managers](/concepts/io-management/io-managers) enable fully overriding this behavior and storing asset contents in any way you wish - e.g. writing them as tables in a database or as objects in a cloud object store. Dagster also provides built-in IO managers that pickle assets to AWS S3 (<PyObject module="dagster_aws.s3" object="s3_pickle_io_manager" />), Azure Blob Storage (<PyObject module="dagster_azure.adls2" object="adls2_pickle_io_manager" />), and GCS (<PyObject module="dagster_gcp.gcs" object="gcs_pickle_io_manager" />), or you can write your own.
+
+To apply an IO manager to a set of assets, you can use <PyObject object="with_resources" />:
+
+```python file=/concepts/assets/asset_io_manager.py startafter=start_marker endbefore=end_marker
+from dagster_aws.s3 import s3_pickle_io_manager, s3_resource
+
+from dagster import asset, with_resources
+
+
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+
+
+assets_with_io_manager = with_resources(
+    [upstream_asset, downstream_asset],
+    resource_defs={"io_manager": s3_pickle_io_manager, "s3": s3_resource},
+)
+```
+
+This example also includes `"s3": s3_resource`, because the `s3_pickle_io_manager` depends on an s3 resource.
```

Comment:
```suggestion
This example also includes `"s3": s3_resource` because the `s3_pickle_io_manager` depends on an S3 resource.
```

Suggesting we remove an unneeded comma and make `s3` to `S3` since I think this refers to S3 as a proper noun and not an explicit value. Feel free to scrap that if I misunderstood.